tls/s2n_handshake.h

```diff
@@ -155,11 +174,20 @@ struct s2n_handshake {
 #define WITH_SESSION_TICKET         0x20
 #define IS_ISSUING_NEW_SESSION_TICKET( type )   ( (type) & WITH_SESSION_TICKET )
 
+/* A HelloRetryRequest was needed to proceed with the handshake */
+#define HELLO_RETRY_REQUEST         0x80
+
     /* Which handshake message number are we processing */
     int message_number;
 
     /* Set to 1 if the RSA verification failed */
     uint8_t rsa_failed;
+
+    /* Used by the server to track whether a HelloRetryRequest was sent */
+    uint8_t hello_retry_request;
+
+    /* Used by the server to determine whether a HelloRetryRequest is needed */
+    unsigned server_requires_hrr:1;
```

Comment:
What is the difference between these two variables?

Reply:
This gets addressed by the HelloRetry message type in the handshake. This was required in order to rewrite the transcript during the hello retry, but not after it was completed.